From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paolo Sabatino <paolo.sabatino@gmail.com>
Date: Wed, 20 Aug 2025 21:40:48 +0200
Subject: [ARCHEOLOGY] workaround for rockchip gigabit ethernet not able to
 init dma

> X-Git-Archeology: - Revision ae11ad5db511f41448066fd29c1c9f5e661b023c: https://github.com/armbian/build/commit/ae11ad5db511f41448066fd29c1c9f5e661b023c
> X-Git-Archeology:   Date: Wed, 20 Aug 2025 21:40:48 +0200
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: workaround for rockchip gigabit ethernet not able to init dma
> X-Git-Archeology:
> X-Git-Archeology: - Revision 35629f641e8e0464f1cb3544e57528689bd87c1d: https://github.com/armbian/build/commit/35629f641e8e0464f1cb3544e57528689bd87c1d
> X-Git-Archeology:   Date: Fri, 03 Oct 2025 18:36:42 +0200
> X-Git-Archeology:   From: Paolo <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: Bump rockchip64-edge kernel to 6.17 (#8703)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 8360df68703e3c5670a24afd43da42cfc597c75b: https://github.com/armbian/build/commit/8360df68703e3c5670a24afd43da42cfc597c75b
> X-Git-Archeology:   Date: Sun, 12 Oct 2025 18:58:55 +0200
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: rockchip64-6.18: bump 6.17 -> 6.18 via rename
> X-Git-Archeology:
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index 111111111111..222222222222 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -3057,8 +3057,8 @@ static int stmmac_init_dma_engine(struct stmmac_priv *priv)
 
 	ret = stmmac_reset(priv, priv->ioaddr);
 	if (ret) {
-		netdev_err(priv->dev, "Failed to reset the dma\n");
-		return ret;
+		netdev_warn(priv->dev, "Failed to reset the dma, device will work with reduced throughput\n");
+		ret = 0;
 	}
 
 	/* DMA Configuration */
-- 
Armbian

